File: Stepic/Web_technologies/ajax.js
var xhr = new XMLHttpRequest();  //Создаем объект (разный для браузеров)
xhr.open('POST', '/xhr/test.html', true); //опен указывает с каким методом и по какому урлу запрос
// true - значит асинхронный
xhr.onreadystatechange = function() { //Создаем функцию callback 
	if (xhr.readyState == 4) {
		if (xhr.status == 200) {
		alert(xhr.responseText);
		}
	}
};
xhr.send("a=5&b=4"); //Тело запроса

// Использование Ajax при ромоще jQuerry

$.ajax({ $ // вывод метода ajax с параметрами
	url: '/blog/comments/add/', // урл
	type: 'POST', // метод
	data: { post_id: 12, text: 'Занятная идея!' }, // данные
// задаем функции обработчки
}).success(function(data) { // если успешно и вернулся 200
	// получаем объект data (в данном случае продпологаем что это json)
	if (data.status == 'ok') {
		console.log(data.comment_id);	
	}
}).error(function() { // если вернулись 4хх или 5хх
	console.log('http error')
});

File: Stepic/Web_technologies/ajax.py
def comments_list(request):
	post_id = request.GET.get('post_id')
	post = get_object_or_404(Post, post_id)
	comments = paginate(request, post.comments)
	return render(request, 'blog/comments.html', {
		'comments': comments
	})


# HttpResponceAjax
import json

# класс для успешных ответов 
class HttpResponseAjax(HttpResponse):
	def __init__(self, status='ok', **kwargs): 
		kwargs['status'] = status  # Добавляем статус 
		super(HttpResponseAjax, self).__init__( 
			content = json.dumps(kwargs),
			content_type = 'application/json',
		)

# класс для ошибок
class HttpResponseAjaxError(HttpResponseAjax): # наследуем от первого 
	def __init__(self, code, message):
		super(HttpResponseAjaxError, self).__init__(
			status = 'error', code = code, message = message
		)

# использоване HttpResponseAjax
@login_required_ajax # 	Для вьюшки пользователь должен быть авторизован
def comment_add(request):
	form = AddCommentForm(request.POST)
	if form.is_valid():
		comment = form.save()
		return HttpResponseAjax(comment_id=comment.id)
	else:
		return HttpResponseAjaxError(
			code = "bad_params",
			message = form.errors.as_text(), # Не очень хорошо
		)	